include/s3select_json_parser.h

```diff
@@ -0,0 +1,511 @@
+    std::vector < std::pair < std::string, Valuesax>> exactfilter;
+    std::vector <std::string> filter_pattern{};
+    std::vector <std::string> filter_exact{};
+    std::vector <std::string> filter_prefix_suffix{};
+    std::vector <std::string> prefix{};
+    std::vector <std::string> suffix{};
+    bool flag{};
+    Valuesax value;
+    ChunksStreamer stream_buffer;
+    bool init_buffer_stream;
+    rapidjson::Reader reader;
+    std::vector<en_json_elm_state_t> json_element_state;
+    std::string m_result;//debug purpose
+    std::vector<std::string> key_path;
+    std::string m_filter_result{};
+    std::string m_filter_exact_result{};
+
+    JsonParserHandler() : init_buffer_stream(false)
+    {}
+
+    std::string get_key_path()
+    {//for debug
+	std::string res;
+	for(const auto & i: key_path)
+	{
+	  res.append(i);
+	  res.append(std::string("/"));
+	}
+	return res;
+    }
+
+    void clear_keyValue_store() {
+      key_value_store.clear();
+    }
+
+    void clear_filtermap() {
+      filtermap.clear();
+    }
+
+    void clear_exactfilter() {
+      exactfilter.clear();
+    }
+
+    std::vector < std::pair < std::string, Valuesax>> get_keyValue_store() {
+      return key_value_store;
+    }
+
+    std::vector < std::pair < std::string, Valuesax>> get_filtervalue() {
+      return filtermap;
+    }
+
+    std::vector < std::pair < std::string, Valuesax>> get_exactfiltervalue() {
+      return exactfilter;
+    }
+
+    void dec_key_path()
+    {
+      if(json_element_state.back() != ARRAY_STATE)
+	{
+	  if(key_path.size() != 0)
+	    key_path.pop_back();
+	}
+    }
+
+    void push_new_key_value(Valuesax& v)
+    {
+      key_value_store.push_back(std::make_pair(get_key_path(), v));
+      if (flag) {
+        filtermap.push_back(std::make_pair(get_key_path(), v));
+      }
+      if (key_path == filter_exact) {
+        exactfilter.push_back(std::make_pair(get_key_path(), v));
+      }
+      dec_key_path();
+    }
+
+    bool Null() {
+      // at this point should verify against from-clause/where-clause/project. if match then push to scratch-area
+      push_new_key_value(value.Parse(nullptr));
+      return true; }
+
+    bool Bool(bool b) {
+      push_new_key_value(value.Parse(b));
+      return true; }
+
+    bool Int(int i) { 
+      push_new_key_value(value.Parse(i));
+      return true; }
+
+    bool Uint(unsigned u) {
+      push_new_key_value(value.Parse(u));
+      return true; }
+
+    bool Int64(int64_t i) { 
+      push_new_key_value(value.Parse(i));
+      return true; }
+
+    bool Uint64(uint64_t u) { 
+      push_new_key_value(value.Parse(u));
+      return true; }
+
+    bool Double(double d) { 
+      push_new_key_value(value.Parse(d));
+      return true; }
+
+    bool String(const char* str, rapidjson::SizeType length, bool copy) {
+      //TODO use copy
+      push_new_key_value(value.Parse(str));
+      return true;
+    }
+
+    bool Key(const char* str, rapidjson::SizeType length, bool copy) {
```

Comment:
`key` function is being called per all JSON element types (array,value,object)

that is the location for from-clause match condition , i.e. it should be prefix-match
thus all keys(values,objects,arrays) should be filtered according to prefix match  
request = a.b.c ; extracted-key = a.b.c.d.e is a match
extracted-key = a.b3.c.d.e is not a match

please note that operator== for vector<string> is an exact match.